src/apps/relay/windows/ServiceInstaller.cpp

```diff
@@ -0,0 +1,162 @@
+/*!
+ * Install service
+ * \author Kang Lin <kl222@126.com>
+ * \see https://learn.microsoft.com/windows/win32/services/service-configuration-program-tasks
+ */
+
+#include "ServiceInstaller.h"
+#include <stdio.h>
+#include <strsafe.h>
+#include <tchar.h>
+#include <windows.h>
+
+/*!
+ * \brief Install service
+ *
+ * \details Install the current application as a service to the local
+ *          service control manager database.
+ *
+ * \param pszServiceName - the name of the service to be installed
+ * \param pszDisplayName - the display name of the service
+ * \param dwStartType - the service start option. This parameter can be one of
+ *        the following values:
+ *          - SERVICE_AUTO_START
+ *          - SERVICE_BOOT_START
+ *          - SERVICE_DEMAND_START
+ *          - SERVICE_DISABLED
+ *          - SERVICE_SYSTEM_START.
+ * \param pszDependencies - a pointer to a double null-terminated array of null-
+ *     separated names of services or load ordering groups that the system
+ *     must start before this service.
+ * \param pszAccount - the name of the account under which the service runs.
+ *        - local account: "NT AUTHORITY\\LocalService"
+ *        - network account: "NT AUTHORITY\NetworkService"
+ *        - local system account: ".\LocalSystem"
+ *        \see https://learn.microsoft.com/windows/win32/services/service-user-accounts
+ * \param pszPassword - the password to the account name.
+ *
+ * \note If the function fails to install the service, it prints the error
+ *         in the standard output stream for users to diagnose the problem.
+ * \see https://learn.microsoft.com/windows/win32/services/service-configuration-programs
+ * \see https://learn.microsoft.com/windows/win32/services/installing-a-service
+ */
+int InstallService(LPTSTR pszServiceName, LPTSTR pszDisplayName, DWORD dwStartType, LPTSTR pszDependencies,
+                   LPTSTR pszAccount, LPTSTR pszPassword) {
+  SC_HANDLE schSCManager;
+  SC_HANDLE schService;
+  TCHAR szUnquotedPath[MAX_PATH];
+
+  if (!GetModuleFileName(NULL, szUnquotedPath, MAX_PATH)) {
+    printf("Cannot install service (%lu)\n", GetLastError());
```

Comment:
"Cannot install service with error: %lu"

though it might be nice to format the errors as strings, e.g. with this function https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-formatmessage